qp_qiime2/tests/test_qiime2.py

```diff
@@ -82,6 +82,79 @@ def test_rarefy_error(self):
         self.assertIsNone(ainfo)
         self.assertEqual('Rarefaction level too high 200000', msg)
 
+    def test_beta(self):
+        out_dir = mkdtemp()
+        self._clean_up_files.append(out_dir)
+
+        # qiime2 currently only works with rarefied tables so we need to
+        # rarefy it
+        params = {'p-sampling-depth': 10, 'i-table': 5}
+        data = {'user': 'demo@microbio.me',
+                'command': dumps(['qiime2', qiime2_version, 'Rarefy']),
+                'status': 'running',
+                'parameters': dumps(params)}
+        jid = self.qclient.post('/apitest/processing_job/', data=data)['job']
+        success, ainfo, msg = rarefy(self.qclient, jid, params, out_dir)
+        data = {'filepaths': dumps(ainfo[0].files), 'type': "BIOM",
+                'name': "Rarefied biom", 'analysis': 1, 'data_type': '16S'}
+        reply = self.qclient.post('/apitest/artifact/', data=data)
+        aid = reply['artifact']
+
+        # # actually test non phylogenetic beta diversity
+        params = {
+            'i-table': aid, 'p-metric': 'euclidean',
+            'i-tree': None}
+        data = {'user': 'demo@microbio.me',
+                'command': dumps(['qiime2', qiime2_version, 'beta_diversity']),
+                'status': 'running',
+                'parameters': dumps(params)}
+        jid = self.qclient.post('/apitest/processing_job/', data=data)['job']
+        success, ainfo, msg = beta_diversity(
+            self.qclient, jid, params, out_dir)
+        self.assertEqual('', msg)
```

Comment:
Change the order of the parameters -> `self.assertEqual(msg, '')`. We typically use first observed, second expected.